grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1121,9 +1121,9 @@ class StockMovementService {
                         code: it.code,
                         name: it.name,
                         lotNumber: it.lotNumber,
-                        expirationDate: new Date(it.expirationDate),
+                        expirationDate: it.expirationDate ? new Date(it.expirationDate) : null,
                         binLocation: it.binLocation,
-                        quantity: Integer.parseInt(it.quantity)
+                        quantity: it.quantity.isNumber() ? Integer.parseInt(it.quantity) : null
```

Comment:
Do we want to return a validation error if it's not a number? Or is it ok to handle the quantity as null? I think the distinction is that if null eventually becomes 0 and 0 is a valid quantity, then it might be confusing to the user who entered accidentially entered "100 ea" to have no picklist items for that line. 

I'm looking at this in a vacuum so I don't have a good sense of the expected behavior or next steps but I would look downstream to see what behavior we've implemented (understanding that the behavior might change in the future) and make sure that we handle it as explicitly as possible, rather than coercsing input to something the user may not have intended.